node/server.js

```diff
@@ -2297,59 +2297,52 @@ function monster_hunt_logic(player, monster) {
 	}
 }
 
-function calculate_monster_score(player, monster, share) {
-	var score = min(1, monster.mult * 2.2);
-	var divider = 1;
-	if (!share) {
-		share = 0;
-	}
-	if (monster.cooperative) {
-		divider = 2;
-	}
-	for (var id in players) {
-		var current = players[id];
-		if (current.id == player.id) {
-			continue;
+function calculate_monster_score(player, monster, coopShare = 0) {
+	// NOTE: `.mult` is usually 1, but will be set on server shutdown to the ratio of remaining HP.
+	//       See: shutdown_routine()
+	let score = Math.min(1, monster.mult * 2.2);
+	const divisor = monster.cooperative ? 2 : 1;
+
+	for (const id in players) {
+		const p = players[id];
+		if (p.id === player.id) {
```

Comment:
Could skip the get call on `id` of p and just use `id` directly given by the for loop

Reply:
https://github.com/earthiverse/adventureland/blob/2f6b1484405f20b086fe813926b01327eca61034/node/server.js#L9674

It might be a different ID?